sdk/core/azure-core-version-tests/src/test/java/com/azure/core/version/tests/MemberNameConverterTests.java

```diff
@@ -0,0 +1,299 @@
+    }
+
+    @Test
+    public void testPropertyNameOnMethodName() throws NoSuchMethodException {
+        class LocalHotel {
+            String hotelName;
+            boolean flag;
+
+            public String getHotelName() {
+                return hotelName;
+            }
+
+            public void setHotelName(String hotelName) {
+                this.hotelName = hotelName;
+            }
+
+            public boolean isFlag() {
+                return flag;
+            }
+
+            public void setFlag(boolean flag) {
+                this.flag = flag;
+            }
+        }
+
+        Method getterM = LocalHotel.class.getDeclaredMethod("getHotelName");
+        assertEquals("hotelName", JACKSON_JSON_SERIALIZER.convertMemberName(getterM));
+        Method setterM = LocalHotel.class.getDeclaredMethod("setHotelName", String.class);
+        assertNull(JACKSON_JSON_SERIALIZER.convertMemberName(setterM));
+        Method getterWithIs = LocalHotel.class.getDeclaredMethod("isFlag");
+        assertEquals("flag", JACKSON_JSON_SERIALIZER.convertMemberName(getterWithIs));
+        Method setterWithIs = LocalHotel.class.getDeclaredMethod("setFlag", boolean.class);
+        assertNull(JACKSON_JSON_SERIALIZER.convertMemberName(setterWithIs));
+    }
+
+    @Test
+    public void testPropertyNameOnMethodNameWithoutGetSet() throws NoSuchMethodException {
+        class LocalHotel {
+            String hotelName;
+
+            public String hotelName1() {
+                return hotelName;
+            }
+
+            public void hotelName2(String hotelName) {
+                this.hotelName = hotelName;
+            }
+        }
+
+        Method getterM = LocalHotel.class.getDeclaredMethod("hotelName1");
+        assertNull(JACKSON_JSON_SERIALIZER.convertMemberName(getterM));
+        Method setterM = LocalHotel.class.getDeclaredMethod("hotelName2", String.class);
+        assertNull(JACKSON_JSON_SERIALIZER.convertMemberName(setterM));
+    }
+
+    @Test
+    public void testPropertyNameOnMethodNameWithGetInMiddle() throws NoSuchMethodException {
+        class LocalHotel {
+            String hotelName;
+            String hotelId;
+            String flag;
+
+            public String hotelgetName() {
+                return hotelName;
+            }
+
+            public String getHotelGetId() {
+                return hotelId;
+            }
+
+            public String isFlag() {
+                return flag;
+            }
+        }
+
+        Method getterM1 = LocalHotel.class.getDeclaredMethod("hotelgetName");
+        Method getterM2 = LocalHotel.class.getDeclaredMethod("getHotelGetId");
+        Method getterM3 = LocalHotel.class.getDeclaredMethod("isFlag");
+
+        assertNull(JACKSON_JSON_SERIALIZER.convertMemberName(getterM1));
+        assertEquals("hotelGetId", JACKSON_JSON_SERIALIZER.convertMemberName(getterM2));
+        assertNull(JACKSON_JSON_SERIALIZER.convertMemberName(getterM3));
```

Comment:
Is this `null` because the return type is not a `boolean` and `is` prefix only works if the method returns a boolean?

Reply:
Yep, or at least that is the default handling